addons/delivery/models/sale_order.py

```diff
@@ -25,6 +34,24 @@ def _compute_amount_total_without_delivery(self):
         delivery_cost = sum([l.price_total for l in self.order_line if l.is_delivery])
         return self.amount_total - delivery_cost
 
+    @api.depends('carrier_id')
+    def _compute_shipping_method(self):
+        for order in self:
+            if hasattr(order.carrier_id, f'{order.carrier_id.delivery_type}_use_locations') and getattr(order.carrier_id, f'{order.carrier_id.delivery_type}_use_locations'):
```

Comment:
would be great to define a compute on carrier for that. To know if pickup point is available or not